portafly/src/components/pages/accounts/ExportAccountsButton.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import { useTranslation } from 'i18n/useTranslation'
+import { ExportIcon } from '@patternfly/react-icons'
+import { CSVDownload } from 'react-csv'
+import { IDeveloperAccount } from 'types'
+import { Button } from '@patternfly/react-core'
+import { render } from '@testing-library/react'
+
+interface Props {
+  data?: IDeveloperAccount[]
+}
+const ExportAccountsButton = ({ data }: Props) => {
+  const { t } = useTranslation('accountsIndex')
+
+  const exportHeaders = [
+    { label: 'ID', key: 'id' },
+    { label: t('accounts_table.admin_header'), key: 'admin_name' },
+    { label: t('accounts_table.group_header'), key: 'org_name' },
+    { label: t('accounts_table.state_header'), key: 'state' },
+    { label: t('accounts_table.applications_header'), key: 'apps_count' },
+    { label: t('accounts_table.created_header'), key: 'created_at' },
+    { label: t('accounts_table.updated_header'), key: 'update_at' }
+  ]
+
+  return (
+    <Button
+      isDisabled={!data}
+      variant="link"
+      icon={<ExportIcon />}
+      aria-label={t('export_accounts_button_aria_label')}
+      onClick={() => !!data && render(
```

Comment:
This is very smelly, no offense. Instead of double bang use a casting, since the type is well-known at this point `data as IDeveloperAccount[]`. And I am not at all convinced about using a testing render method... The correct way is to render it as a sibling.

But after running it locally and playing a bit with it, everything points to using <CSVLink> instead :/ I think it's the lesser evil after all.

Reply:
Shit. It really smells, I didn't want to use that particular one TBH. I think it was auto-completed and auto-imported by VS. Thanks for spotting it!

Reply:
it doesn't smell, it stinks! a strong offensive odor right there

Reply:
I think the lesser evil here would be to use the `CSVLink` component

Reply:
> I think the lesser evil here would be to use the `CSVLink` component

That's what I meant exactly! 😅  But I realize now I left a gap:
> [...] everything points to using _`CSVLink`_  instead

Reply:
<img width="482" alt="Screen Shot 2020-06-17 at 09 04 26" src="https://user-images.githubusercontent.com/11672286/84866104-8eb55000-b079-11ea-9cc0-cdffb20a290b.png">
Jeez, it's there but it's invisible. I forgot to close the tag I guess!